nova/tests/unit/scheduler/weights/test_weights_resize_same_host.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 OpenStack Foundation
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+Tests for scheduler prefer-resize-to-same-host weigher.
+"""
+from nova import objects
+from nova.scheduler import weights
+from nova.scheduler.weights import resize_same_host
+from nova import test
+from nova.tests.unit.scheduler import fakes
+
+
+class PreferSameHostOnResizeWeigherTestCase(test.NoDBTestCase):
+    def setUp(self):
+        super(PreferSameHostOnResizeWeigherTestCase, self).setUp()
+        self.weight_handler = weights.HostWeightHandler()
+        self.weighers = [resize_same_host.PreferSameHostOnResizeWeigher()]
+
+    def _prepare_instances_and_hosts(self):
+        flavor_big = objects.Flavor(name='big', memory_mb=1024 * 512,
+                                    extra_specs={})
+        flavor_small = objects.Flavor(name='small', memory_mb=1024 * 256,
+                                      extra_specs={})
+        instances = [
+            objects.Instance(uuid='00000000-0000-4000-0000-000000000001',
+                             flavor=flavor_small),
+            objects.Instance(uuid='00000000-0000-4000-0000-000000000002',
+                             flavor=flavor_big),
+        ]
+        hosts = [
+            fakes.FakeHostState('same_host', 'n1', {}, instances),
+            fakes.FakeHostState('other_host', 'n2', {}),
+        ]
+        request_specs = [
+            objects.RequestSpec(instance_uuid=instances[0].uuid,
+                                flavor=flavor_big),
+            objects.RequestSpec(instance_uuid=instances[0].uuid,
+                                flavor=flavor_small),
+        ]
+        return instances, hosts, request_specs
+
+    def test_prefer_resize_to_same_host(self):
+        self.flags(prefer_same_host_resize_weight_multiplier=1.0,
+                   group='filter_scheduler')
+        instances, hosts, request_specs = self._prepare_instances_and_hosts()
+        weighed_hosts = self.weight_handler.get_weighed_objects(
+            self.weighers, hosts, request_specs[0])
+        self.assertEqual(1.0, weighed_hosts[0].weight)
+        self.assertEqual(0.0, weighed_hosts[-1].weight)
```

Comment:
What about all the hosts in between? Seeing a `-1` I expect to have more hosts. If you know there are just two, I'd use `[0]` and `[1]`.

Reply:
This says what I want to say more. The first host has a value of 1, whereas the last host has to have a 0.

It's basically coding for change in case future test-case changes introduce more hosts for some reason. If that is the case, this code doesn't need to be touched.

Reply:
If you want to say, that only one host should have `1.0`, you should do a `sum(h.weight for h in weighed_hosts[1:]) == 0` instead. Because only checking the last host in the list doesn't make sense to me.

Reply:
Okay, that makes sense. I don't like the sum-thing, because that feels like asking for (admittedly highly unlikely) `[1, 0, -1]` trouble, but I will change it so it checks the second host again, since weighing guarantees ordering, and all I want to check is indeed that only one host got the 1.0, yes.